IllegalStateException: TorManager$broadcastReceiver$.onReceive

Samsung Galaxy A31 Android 11 (SDK 30)

```ruby
java.lang.RuntimeException: 
  at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args (LoadedApk.java:1676)
  at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run (Unknown Source:2)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8550)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1139)
Caused by: java.lang.IllegalStateException: 
  at org.onionshare.android.tor.TorManager$broadcastReceiver$1.onReceive (TorManager.kt:64)
  at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args (LoadedApk.java:1666)
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -3,7 +3,7 @@ plugins {
     id 'kotlin-android'
     id 'kotlin-kapt'
     id 'dagger.hilt.android.plugin'
-    id 'org.jlleitschuh.gradle.ktlint' version '10.2.0'
+    id 'org.jlleitschuh.gradle.ktlint' version '10.2.1'
 }
 
 static def versionCodeEpoch() {
@@ -91,19 +91,19 @@ dependencies {
     implementation 'androidx.core:core-ktx:1.7.0'
     implementation 'androidx.appcompat:appcompat:1.4.1'
     implementation 'androidx.activity:activity-compose:1.4.0'
-    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.4.0'
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.4.1'
     implementation 'com.google.android.material:material:1.5.0'
     implementation "androidx.compose.ui:ui:$compose_version"
     implementation "androidx.compose.material:material:$compose_version"
     implementation "androidx.compose.material:material-icons-extended:$compose_version"
     implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
-    implementation "androidx.navigation:navigation-compose:2.4.1"
+    implementation "androidx.navigation:navigation-compose:2.4.2"
     implementation "androidx.profileinstaller:profileinstaller:1.2.0-alpha02"
 
     implementation "com.google.dagger:hilt-android:$hilt_version"
     kapt "com.google.dagger:hilt-compiler:$hilt_version"
 
-    implementation 'info.guardianproject:tor-android:0.4.6.8'
+    implementation 'info.guardianproject:tor-android:0.4.6.10'
     implementation 'info.guardianproject:jtorctl:0.4.5.7'
 
     def ktor_version = '1.6.5'

--- a/app/src/main/java/org/onionshare/android/tor/TorManager.kt
+++ b/app/src/main/java/org/onionshare/android/tor/TorManager.kt
@@ -25,6 +25,7 @@ import org.onionshare.android.server.PORT
 import org.slf4j.LoggerFactory.getLogger
 import org.torproject.jni.TorService
 import org.torproject.jni.TorService.ACTION_STATUS
+import org.torproject.jni.TorService.EXTRA_SERVICE_PACKAGE_NAME
 import org.torproject.jni.TorService.EXTRA_STATUS
 import java.io.File
 import java.io.FileInputStream
@@ -57,14 +58,13 @@ class TorManager @Inject constructor(
          * Attention: This gets executes on UI Thread
          */
         override fun onReceive(context: Context, i: Intent) {
+            val intentPackageName = i.getStringExtra(EXTRA_SERVICE_PACKAGE_NAME)
+            if (intentPackageName != context.packageName) return // this intent was not for us
             when (i.getStringExtra(EXTRA_STATUS)) {
                 TorService.STATUS_STARTING -> LOG.debug("TorService: Starting...")
                 TorService.STATUS_ON -> {
                     LOG.debug("TorService: Started")
-                    // TODO remove if condition and replace LOG.error with error() once #29 is resolved
-                    if (File(getControlPath()).exists()) {
-                        startLatch?.countDown() ?: LOG.error("startLatch was null when Tor started.")
-                    }
+                    startLatch?.countDown() ?: LOG.error("startLatch was null when Tor started.")
                 }
                 // FIXME When we stop unplanned, we need to inform the ShareManager
                 //  that we stopped, so it can clear its state up, stopping webserver, etc.

--- a/build.gradle
+++ b/build.gradle
@@ -2,7 +2,7 @@ buildscript {
     ext {
         kotlin_version = '1.6.10'
         hilt_version = '2.40'
-        compose_version = '1.1.0'
+        compose_version = '1.1.1'
     }
     repositories {
         google()

